### PR TITLE
ref: a real start

### LINKED ISSUE #155: ref: full command reference
Follow-up to #154 <- https://github.com/iterative/cml/issues/361 because copy-pasting from CLI `--help` into https://cml.dev/doc/ref is far from ideal.

### PR DESCRIPTION
**Pull request by casperdcl**
- fixes #155

### CODE REVIEW

File: content/docs/ref/runner.md [hunk 1]
```diff
@@ -11,46 +11,38 @@ compute provider or locally on-premise).
...
+- `--cloud-hdd-size=<...>`: Disk storage size in GB.
+- `--cloud-ssh-private=<...>`: Private SSH RSA key [default: *auto-generate
+  throwaway key*].
+- `--cloud-spot`: Request a preemptible spot instance.
+- `--cloud-spot-price=<...>`: Maximum spot instance USD bidding price, [default:
+  *current price*].
+- `--cloud-startup-script=<...>`: Run the provided
+  [Base64](https://linux.die.net/man/1/base64)-encoded Linux shell script during
+  the instance initialization.
+- `--cloud-aws-security-group=<...>`: AWS security group.
```
Comment (casperdcl) [inline content/docs/ref/runner.md:45]: TODO: add crossrefs for `--cloud-*` commands to https://registry.terraform.io/providers/iterative/iterative/latest/docs/resources/task#attribute-reference

File: content/docs/ref/pr.md [hunk 2]
```diff
@@ -17,12 +17,11 @@ preventing an infinite chain of runs.
 
 Any [generic option](/doc/ref) in addition to:
 
-```
---md          Output in markdown format [](url).                     [boolean]
---remote      Sets git remote.                    [string] [default: "origin"]
---user-email  Sets git user email.   [string] [default: "olivaw@iterative.ai"]
---user-name   Sets git user name.            [string] [default: "Olivaw[bot]"]
-```
+- `--md`: Produce output in markdown format (`[Pull/Merge Request](url)`).
```
Comment (casperdcl) [inline content/docs/ref/pr.md:20]: `cml pr` has a `--md` flag but no `-f, --file` flag. We should either add one, or remove `-f, --file` from `cml publish` & `cml tensorboard-dev`

Comment (daavoo) [reply content/docs/ref/pr.md:20]: Maybe add an example of the output of `cml pr --md` to show how it is different from plain `cml pr`? 

It doesn't seem intuitive to me what the difference would be without an example:

https://github.com/iterative/cml/blob/master/src/cml.js#L339-L343

Comment (daavoo) [reply content/docs/ref/pr.md:20]: Unless `-f` does some extra logic that you can't achieve with plain output redirection, I would drop `--file` options from other commands (maybe adding example instead like `cml cmd --md > output.md`)

File: content/docs/ref/runner.md (hunk 1 above)
Comment (0x2b3bfa0) [reply content/docs/ref/runner.md:48]: The latest version of the provider supports the same `--cloud-type` and `--cloud-region` values on all the resources. Adding those references would be great.

File: content/docs/self-hosted-runners.md [hunk 3]
```diff
@@ -148,19 +148,23 @@ For example, `docker://iterativeai/cml:0-dvc2-base1-gpu`, or
 The `cml runner` command supports many options (see the
 [command reference](/doc/ref/runner)). Notable options are:
 
-- `--labels`: One or more comma-delimited labels (e.g. "cml,gpu").
-- `--idle-timeout`: Seconds to wait for jobs before shutting down.
-- `--single`: Terminate after running a single job.
+- `--labels=<...>`: One or more (comma-delimited) labels (e.g. `cml,gpu`).
+- `--idle-timeout=<seconds>`: Seconds to wait for jobs before shutting down.
+- `--single`: Shutdown runner after one workflow run.
```
Comment (0x2b3bfa0) [inline content/docs/self-hosted-runners.md:153]:
```suggestion
- `--single`: Shutdown runner after one workflow run.
```

Even “quit” or “exit” perhaps?

Comment (casperdcl) [reply content/docs/self-hosted-runners.md:153]: Neither "quit" nor "exit" sound quite right (PS: I see an empty diff in your suggested change). How about

```suggestion
- `--single`: Terminate runner after one workflow run.
```

File: content/docs/ref/pr.md (hunk 2 above)
Comment (casperdcl) [reply content/docs/ref/pr.md:20]: removed

File: content/docs/self-hosted-runners.md (hunk 3 above)
Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md]: > _I see an empty diff in your suggested change_

Agh! Edited to remove the “suggestion” block

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md]: “Terminate” sounds a bit like AWS EC2 instances, but it's not necessarily tied to those, much less for on-premises scenarios. What about “remove” or “unregister”?

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md]: When providing this option, runners will take a single job and exit, where exiting implies unregistering themselves.

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md]: 🚲 🅿️ 🤷🏼‍♂️

**Review by 0x2b3bfa0 — APPROVED** [review]

File: content/docs/self-hosted-runners.md (hunk 3 above)
Comment (casperdcl) [reply content/docs/self-hosted-runners.md]: seems important though... Scripts/jobs can exit/quit, but instances/machines can terminate/shut down. Seems more important to mention the termination.

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md]: Termination is exclusive to `cml runner --cloud`

**Pull request merged**